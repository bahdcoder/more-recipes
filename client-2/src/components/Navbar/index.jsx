import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar navbar-expand-lg navbar-custom">
    <Link className="navbar-brand" to="/">
      <img src={`${process.env.PUBLIC_URL}/img/logo.png`} className="navbar-logo-img mr-2" alt=''/>
      More-recipes
    </Link>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span className="navbar-toggler-icon" />
    </button>
    <div className="collapse navbar-collapse" id="navbarSupportedContent">
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <a href="" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></a>
        </li>
        <li className="nav-item mr-3">
          <a className="nav-link" href="">Recipes <span className="sr-only">(current)</span></a>
        </li>
        <form className="form-inline my-2 my-lg-0">
          <input className="form-control mr-sm-2" placeholder="Search" aria-label="Search" type="text" />
        </form>
      </ul>
    </div>
  </nav>
);

export default Navbar;
