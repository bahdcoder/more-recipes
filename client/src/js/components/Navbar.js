import React from 'react';
import { NavLink } from 'react-router-dom';

export default class NavBar extends React.Component {
  render() {
    return (
      <nav className="navbar navbar-expand-lg navbar-custom">
        <NavLink className="navbar-brand" to="/">
          <img src="../../../assets/img/logo.png" className="navbar-logo-img mr-2" alt />
          More-recipes
        </NavLink>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <NavLink to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></NavLink>
            </li>
            <li className="nav-item mr-3">
              <NavLink to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></NavLink>
            </li>
            <form className="form-inline my-2 my-lg-0">
              <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search" />
            </form>
          </ul>
        </div>
      </nav>
    );
  }
}