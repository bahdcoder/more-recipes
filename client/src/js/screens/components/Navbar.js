import React from 'react';
import { Link } from 'react-router';
import Gravatar from 'react-gravatar';

export default class NavBar extends React.Component {
  render() {
    let authUser = this.props.authUser;
    let navbarUser = '';
    if (authUser) {
      navbarUser = (
        <li className="nav-item dropdown">
          <a className="nav-link dropdown-toggle" href="javascript:;" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
          <Gravatar className="mr-2 navbar-avatar-img" email={this.props.authUser.user.email}/>
                  Hey, {this.props.authUser.user.name}
          </a>
          <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
              <a className="dropdown-item" href="profile.html">My profile</a>
              <a className="dropdown-item" href="favorites.html">My favorites</a>
              <a className="dropdown-item" href="settings.html">Notification settings</a>
              <a className="dropdown-item" 
                 href="javascript:;" 
                 onClick={async () => {
                   await this.props.signOut();
                   this.props.router.push('/');
                 }}>Sign out</a>
          </div>
      </li>
      );
    }
    return (
      <nav className="navbar navbar-expand-lg navbar-custom">
        <Link className="navbar-brand" to="/">
          <img src="../../../assets/img/logo.png" className="navbar-logo-img mr-2" />
          More-recipes
        </Link>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></Link>
            </li>
            <li className="nav-item mr-3">
              <Link to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></Link>
            </li>
            <form className="form-inline my-2 my-lg-0">
              <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search" />
            </form>
            {navbarUser}
          </ul>
        </div>
      </nav>
    );
  }
}