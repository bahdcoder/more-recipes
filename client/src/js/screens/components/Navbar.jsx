/* eslint-disable no-script-url */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import Gravatar from 'react-gravatar';

import logo from './../../../assets/img/logo.png';

/**
 * The navbar component
 * @param {object} props
 * @returns {jsx} jsx
 */
const Navbar = (props) => {
  const { authUser } = props;
  let navbarUser;
  if (authUser) {
    navbarUser = (
      <li className="nav-item dropdown">
        <a className="nav-link dropdown-toggle" id="navbarDropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
          <Gravatar className="mr-2 navbar-avatar-img" email={props.authUser.user.email} />
          Hey, {props.authUser.user.name}
        </a>
        <div className="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
          <Link className="dropdown-item" to={`/user/${props.authUser.user.id}`}>My profile</Link>
          <Link id="myRecipes" className="dropdown-item" to={`/user/${props.authUser.user.id}/recipes`}>My recipes</Link>
          <Link className="dropdown-item" to="/my/favorites">My favorites</Link>
          <a
            className="dropdown-item logout"
            id="logout"
            onClick={async () => {
              await props.signOut();
              props.router.push('/');
            }}
          >
            Sign out
          </a>
        </div>
      </li>
    );
  }

  return (
    <nav className="navbar navbar-expand-lg navbar-custom">
      <Link className="navbar-brand" to="/">
        <img src={logo} className="navbar-logo-img mr-2" alt="" />
        More-recipes
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link to="/recipes/create" className="nav-link create-recipe-link">Create recipe <span className="sr-only">(current)</span></Link>
          </li>
          <li className="nav-item mr-3">
            <Link to="/recipes" className="nav-link">Recipes <span className="sr-only">(current)</span></Link>
          </li>
          <form
            className="form-inline my-2 my-lg-0"
            onSubmit={(event) => {
              event.preventDefault();
              if (props.location.pathname === '/recipes') {
                props.triggerGetRecipesCatalog();
              } else {
                props.router.push(`/recipes?query=${props.location.query.query || ''}`);
              }
            }}
          >
            <input
              className="form-control mr-sm-2"
              value={props.location.query.query || ''}
              onChange={(event) => {
                props.changeRouterQueryParams('query', event.target.value, props.location);
              }}
              type="text"
              placeholder="Search recipes"
              aria-label="Search"
            />
          </form>
          {navbarUser}
        </ul>
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  authUser: PropTypes.objectOf(PropTypes.any),
  signOut: PropTypes.func.isRequired,
  location: PropTypes.objectOf(PropTypes.any).isRequired,
  triggerGetRecipesCatalog: PropTypes.func.isRequired,
  changeRouterQueryParams: PropTypes.func.isRequired,
  router: PropTypes.objectOf(PropTypes.any).isRequired
};

Navbar.defaultProps = {
  authUser: null
};

export default Navbar;
