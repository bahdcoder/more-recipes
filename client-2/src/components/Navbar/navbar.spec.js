import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Navbar from './index';

it('The navbar renders without crashing', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
