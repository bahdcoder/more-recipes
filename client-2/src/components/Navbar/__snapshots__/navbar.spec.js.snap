// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The footer renders without crashing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-custom"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      className="navbar-logo-img mr-2"
      src="/img/logo.png"
    />
    More-recipes
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link create-recipe-link"
          href=""
        >
          Create recipe 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        className="nav-item mr-3"
      >
        <a
          className="nav-link"
          href=""
        >
          Recipes 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <form
        className="form-inline my-2 my-lg-0"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="text"
        />
      </form>
    </ul>
  </div>
</nav>
`;

exports[`The navbar renders without crashing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-custom"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      className="navbar-logo-img mr-2"
      src="/img/logo.png"
    />
    More-recipes
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link create-recipe-link"
          href=""
        >
          Create recipe 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        className="nav-item mr-3"
      >
        <a
          className="nav-link"
          href=""
        >
          Recipes 
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <form
        className="form-inline my-2 my-lg-0"
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="text"
        />
      </form>
    </ul>
  </div>
</nav>
`;
